File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/elements/IRelation.java
package edu.kit.kastel.lissa.sketches.model.elements;

import org.eclipse.collections.api.list.ImmutableList;

import edu.kit.kastel.lissa.sketches.model.ISketch;
import edu.kit.kastel.lissa.sketches.model.impl.Relation;
import edu.kit.kastel.lissa.sketches.model.types.SketchRelationTypeMapping;
import edu.kit.kastel.lissa.sketches.model.types.SketchRelationTypes;

@SketchRelationTypeMapping(type = SketchRelationTypes.UNKNOWN, implementation = Relation.class)
public interface IRelation extends ISketchElement {
	SketchRelationTypes getCurrentInterpretation();

	void addToAssociation(IBox element);

	void delFromAssociation(IBox element);

	ImmutableList<IBox> getConnectedElements();

	void reloadElementMappings(ISketch sketch);
}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/elements/IRelation.kt
package edu.kit.kastel.lissa.sketches.model.elements

import edu.kit.kastel.lissa.sketches.model.ISketch
import edu.kit.kastel.lissa.sketches.model.impl.RelationImpl
import edu.kit.kastel.lissa.sketches.model.types.SketchElementType
import edu.kit.kastel.lissa.sketches.model.types.SketchRelationTypeMapping
import edu.kit.kastel.lissa.sketches.model.types.SketchRelationTypes

@SketchRelationTypeMapping(type = SketchRelationTypes.UNKNOWN, implementation = RelationImpl::class)
interface IRelation : ISketchElement {
    fun currentInterpretation(): SketchRelationTypes

    fun connectedElements(): List<IBox>
    fun addToAssociation(element: IBox)
    fun delFromAssociation(element: IBox)

    fun reloadElementMappings(sketch: ISketch)

    override fun elementType(): SketchElementType = SketchElementType.RELATION
}
